lib/resty/upstream/healthcheck.lua

```diff
@@ -289,85 +294,66 @@ local function check_peer(ctx, id, peer, is_backup)
     sock:close()
 end
 
-local function check_peer_range(ctx, from, to, peers, is_backup)
-    for i = from, to do
-        check_peer(ctx, i - 1, peers[i], is_backup)
-    end
-end
-
 local function check_peers(ctx, peers, is_backup)
-    local n = #peers
+    local n = 0
+    for _ in pairs(peers) do n = n + 1 end
```

Comment:
Because peers is a hash and not an array, after the first upstream, for the next upstream #peers returns 0.